admin.go

```diff
@@ -474,6 +479,8 @@ func replaceRemoteAdminServer(ctx Context, cfg *Config) error {
 		return err
 	}
 
+	serverMu.Lock()
+	defer serverMu.Unlock()
```

Comment:
Same as above, we probably shouldn't use defer here.

Reply:
Makes sense, updated in f7b1ae558b11b7f61a2dcc6b9ec19f19d57c8265